compiler/rustc_span/src/lib.rs

```diff
@@ -658,6 +658,9 @@ impl Span {
             .outer_expn_data()
             .allow_internal_unstable
             .map_or(false, |features| features.iter().any(|&f| f == feature))
+            // The enum_as_repr macro is statically added at parse-time regardless of feature
+            // enablement, and no-ops itself if the feature isn't enabled.
+            || (feature == sym::enum_as_repr && self.is_dummy())
```

Comment:
This would have to change if we change the approach above to use a later pass for the attr.